modules/core/src/lpsolver.cpp

```diff
@@ -285,11 +300,16 @@ static int inner_simplex(Mat_<double>& c, Mat_<double>& b,double& v,vector<int>&
         int row_it=0;
         MatIterator_<double> min_row_ptr=b.begin();
         for(MatIterator_<double> it=b.begin();it!=b.end();it+=b.cols,row_it++){
-            double myite=0;
-            //check constraints, select the tightest one, reinforcing Bland's rule
-            if((myite=it[e])>0){
+            if (fabs(it[b.cols-1]) <= SOLVELP_EPS_REL * fabs(it[b.cols-1]) + SOLVELP_EPS_ABS)
```

Comment:
The check is equivalent to `(1-SOLVELP_EPS_REL)*fabs(it[b.cols-1]` <= SOLVELP_EPS_ABS` or `fabs(it[b.cols-1] < SOLVELP_EPS_ABS/0.2`. I do not think that it's expected.